autoload/lsp/utils/text_edit.vim

```diff
@@ -143,3 +135,17 @@ function! s:_switch(path) abort
   endif
 endfunction
 
+"
+" delete
+"
+function! s:delete(bufnr, start, end) abort
+  if exists('*deletebufline')
+      call deletebufline(a:bufnr, a:start, a:end)
+  else
+      let l:foldenable = &foldenable
+      setlocal nofoldenable
+      execute printf('%s,%sdelete _', a:start, a:end)
+      let &foldenable = l:foldenable
+  endif
```

Comment:
`setlocal nofoldenable` is needed to delete folded lines.

I was found this technique on `help winsaveview`.